### PR TITLE
[H2] Support event sigv4 signing

### PR DESCRIPTION
**Pull request by cjyclaire**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

this PR separate `aws-sigv4` gem and `aws-eventstream` gem diff for easier review of #1942 
after review complete, this PR will be merged before #1942

### CODE REVIEW

**Comment by cjyclaire** [comment]
This PR is ready for review

File: gems/aws-sigv4/lib/aws-sigv4/signer.rb [hunk 1]
```diff
@@ -243,6 +244,24 @@ def sign_request(request)
         )
       end
 
+      # Signs a event and returns signature headers and prior signature
```
Comment (awood45) [inline gems/aws-sigv4/lib/aws-sigv4/signer.rb]: Let's flesh out this documentation with documentation of parameters and the return value. This method is doing a lot of stuff (for one example, explaining how we're changing the encoding at the end with the unpacking).

File: gems/aws-sigv4/lib/aws-sigv4/signer.rb [hunk 2]
```diff
@@ -375,6 +394,19 @@ def string_to_sign(datetime, canonical_request)
         ].join("\n")
       end
 
+      def event_string_to_sign(datetime, headers, payload, prior_signature, encoder)
+        encoded_headers = encoder.encode_headers(
+          Aws::EventStream::Message.new(headers: headers, payload: payload)).read
```
Comment (awood45) [inline gems/aws-sigv4/lib/aws-sigv4/signer.rb]: indentation here is a bit hard to follow, I'd at least most the last `).read` down to its own line.

File: gems/aws-sigv4/spec/signer_spec.rb [hunk 3]
```diff
@@ -270,6 +271,29 @@ module Sigv4
 
       end
 
+      context '#sign_event' do
```
Comment (awood45) [inline gems/aws-sigv4/spec/signer_spec.rb]: Are there any partial event receive cases we need to handle here, or is that abstracted out before we get to this phase? Defensive coding here is a good idea.

Comment (cjyclaire) [reply gems/aws-sigv4/spec/signer_spec.rb]: Yeah, currently no. This event signature v4 signing is currently used for event sending only and received events only requires eventstream decoding.

**Comment by cjyclaire** [comment]
Ready for another review  🎉
